File: 07_app_development/MyProject/libtsparser/src/main/java/com/tosmart/xiexuming/libtsparser/table/Service.java
package com.tosmart.xiexuming.libtsparser.table;


import androidx.annotation.NonNull;

import com.tosmart.xiexuming.libtsparser.descriptor.Descriptor;

import java.util.ArrayList;


/**
 * @author xiexuming
 */
public class Service {
    private int serviceId;
    private int EITScheduleFlag;
    private int EITPresentFollowingFlag;
    private int runningStatus;
    private int freeCAMode;
    private int descriptorsLoopLength;
    private ArrayList<Descriptor> Descriptor;

    @NonNull
    @Override
    public String toString() {
        return "Service{" +
                "serviceId=0x" + Integer.toHexString(serviceId) +
                ", EITScheduleFlag=" + EITScheduleFlag +
                ", EITPresentFollowingFlag=" + EITPresentFollowingFlag +
                ", runningStatus=" + runningStatus +
                ", freeCAMode=" + freeCAMode +
                ", descriptorsLoopLength=" + descriptorsLoopLength +
                ", serviceDescriptor=" + Descriptor +
                '}';
    }

    public ArrayList<com.tosmart.xiexuming.libtsparser.descriptor.Descriptor> getDescriptor() {
        return Descriptor;
    }

    public void setDescriptor(ArrayList<com.tosmart.xiexuming.libtsparser.descriptor.Descriptor> descriptor) {
        Descriptor = descriptor;
    }

    public int getServiceId() {
        return serviceId;
    }

    public void setServiceId(int serviceId) {
        this.serviceId = serviceId;
    }

    public int getEITScheduleFlag() {
        return EITScheduleFlag;
    }

    public void setEITScheduleFlag(int EITScheduleFlag) {
        this.EITScheduleFlag = EITScheduleFlag;
    }

    public int getEITPresentFollowingFlag() {
        return EITPresentFollowingFlag;
    }

    public void setEITPresentFollowingFlag(int EITPresentFollowingFlag) {
        this.EITPresentFollowingFlag = EITPresentFollowingFlag;
    }

    public int getRunningStatus() {
        return runningStatus;
    }

    public void setRunningStatus(int runningStatus) {
        this.runningStatus = runningStatus;
    }

    public int getFreeCAMode() {
        return freeCAMode;
    }

    public void setFreeCAMode(int freeCAMode) {
        this.freeCAMode = freeCAMode;
    }

    public int getDescriptorsLoopLength() {
        return descriptorsLoopLength;
    }

    public void setDescriptorsLoopLength(int descriptorsLoopLength) {
        this.descriptorsLoopLength = descriptorsLoopLength;
    }
}

File: newpaserapp/libtosmartparse/src/main/java/com/example/libtosmartparse/table/Service.kt
package com.example.libtosmartparse.table

import com.example.libtosmartparse.descriptor.Descriptor

/**
 * @author xxm
 * 2022/7/17
 **/
class Service {
    private var serviceId = 0
    private var EITScheduleFlag = 0
    private var EITPresentFollowingFlag = 0
    private var runningStatus = 0
    private var freeCAMode = 0
    private var descriptorsLoopLength = 0
    private var Descriptor: ArrayList<Descriptor>? = null
    override fun toString(): String {
        return "Service{" +
                "serviceId=0x" + Integer.toHexString(serviceId) +
                ", EITScheduleFlag=" + EITScheduleFlag +
                ", EITPresentFollowingFlag=" + EITPresentFollowingFlag +
                ", runningStatus=" + runningStatus +
                ", freeCAMode=" + freeCAMode +
                ", descriptorsLoopLength=" + descriptorsLoopLength +
                ", serviceDescriptor=" + Descriptor +
                '}'
    }

    fun getDescriptor(): ArrayList<Descriptor>? {
        return Descriptor
    }

    fun setDescriptor(descriptor: ArrayList<Descriptor>?) {
        Descriptor = descriptor
    }

    fun getServiceId(): Int {
        return serviceId
    }

    fun setServiceId(serviceId: Int) {
        this.serviceId = serviceId
    }

    fun getEITScheduleFlag(): Int {
        return EITScheduleFlag
    }

    fun setEITScheduleFlag(EITScheduleFlag: Int) {
        this.EITScheduleFlag = EITScheduleFlag
    }

    fun getEITPresentFollowingFlag(): Int {
        return EITPresentFollowingFlag
    }

    fun setEITPresentFollowingFlag(EITPresentFollowingFlag: Int) {
        this.EITPresentFollowingFlag = EITPresentFollowingFlag
    }

    fun getRunningStatus(): Int {
        return runningStatus
    }

    fun setRunningStatus(runningStatus: Int) {
        this.runningStatus = runningStatus
    }

    fun getFreeCAMode(): Int {
        return freeCAMode
    }

    fun setFreeCAMode(freeCAMode: Int) {
        this.freeCAMode = freeCAMode
    }

    fun getDescriptorsLoopLength(): Int {
        return descriptorsLoopLength
    }

    fun setDescriptorsLoopLength(descriptorsLoopLength: Int) {
        this.descriptorsLoopLength = descriptorsLoopLength
    }
}
